block-streamer/src/block_stream.rs

```
@@ -269,7 +269,7 @@ async fn process_bitmap_indexer_blocks(
 
     let mut last_published_block_height: u64 = start_block_height;
 
-    while let Some(Ok(block_height)) = matching_block_heights.next().await {
+    while let Some(block_height) = matching_block_heights.next().await.transpose()? {
```

The thing here is we could also trace the error and then proceed to Lake. But I felt this was a situation that deserves a fail fast given that the stored `Last Processed Block` will allow for graceful restarts and errors in the backfill process should be fixed immediately.

---

In what case do we get an error? Can we handle it internally with retries? 

I'm on the fence about this. We could log it, monitor it and perhaps set up an alert, and then fix, all while not causing disruptions to users?

---

Agreed in team meeting to go with a resilient approach where possible. I've changed the code to instead log an error and then proceed to lake.